public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/Editor.tsx

```diff
@@ -52,8 +52,12 @@ function getTemplateVariableOptions() {
 }
 
 // helper function to make it easy to call this from the widget-render-code
-function withTemplateVariableOptions(optionsPromise: Promise<string[]>): Promise<string[]> {
-  return optionsPromise.then((options) => [...getTemplateVariableOptions(), ...options]);
+function withTemplateVariableOptions(optionsPromise: Promise<string[]>, filter?: string): Promise<string[]> {
+  let templateVariableOptions = getTemplateVariableOptions();
+  if (filter && filter !== '') {
```

Comment:
you can remove the `&& filter !== ''), that case is already covered by the truthyness of filter